### PR TITLE
Make AMS API fallback mechanism configurable

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

With this PR the fallback mechanism in case that AMS API is not configured or it isn't working is configurable.

Refactored the amsclient and its usages to make it easier to use a test double.
Refactored unit tests that rely on the fallback mechanism
Added new UT for testing fallback mechanism in case it is needed

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Refactor (refactoring code, removing useless files)
- Unit tests (no changes in the code)

## Testing steps

Tested though unit tests

## Checklist
* [x] `make before_commit` passes
* [x] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [x] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by dpensi — APPROVED** [review]

**Review by Bee-lee — APPROVED** [review]
looks good, pretty elegant 
so just calling `readClusterIDsForOrgID` will take care of it, right?

File: server/server.go [hunk 1]
```diff
@@ -74,14 +74,15 @@ const (
...
-	ErrorChannel      chan error
-	Serv              *http.Server
+	Config                 Configuration
+	InfoParams             map[string]string
+	ServicesConfig         services.Configuration
+	amsClient              amsclient.AMSClient
+	GroupsChannel          chan []groups.Group
+	ErrorFoundChannel      chan bool
+	ErrorChannel           chan error
+	useOrgClustersFallback bool
```
Comment (tisnik) [inline server/server.go:84]: ah should not it be part of `Config`?

Comment (joselsegura) [reply server/server.go:84]: Mmm, and in fact, it is! It is really duplicated. I will push a new commit. Good catch!

**Review by tisnik — APPROVED** [review]
ok

**Pull request merged**